sdk/dotnet/Pulumi/Deployment/Deployment_RegisterResource.cs

```diff
@@ -21,6 +21,10 @@ public partial class Deployment
             var label = $"resource:{name}[{type}]";
             Log.Debug($"Registering resource start: t={type}, name={name}, custom={custom}, remote={remote}");
 
+            if (options.DeletedWith != null && await MonitorSupportsDeletedWith().ConfigureAwait(false)) {
+                throw new Exception("resource option DeletedWith not supported - please update pulumi")
```

Comment:
Small change to wording to match some other "please update" errors we already have.
```suggestion
                throw new Exception(""The Pulumi CLI does not support the DeletedWith option. Please update the Pulumi CLI."")
```
If you could apply that re-wording to the other SDKs as well please.